src/ZoneServer/Network/PacketHandler.cs

```diff
@@ -1632,12 +1633,20 @@ public void CZ_MAP_REVEAL_INFO(IZoneConnection conn, Packet packet)
 				return;
 			}
 
-			var revealedMap = new RevealedMap(mapId, visible, 0);
+			// Check the percentage for validity
+			if (percentage < 0 || percentage > 100)
+			{
+				Log.Warning("CZ_MAP_SEARCH_INFO: User '{0}' tried to update the visibility for map '{1}' beyond an acceptable percentage.", conn.Account.Name, mapId);
+				return;
+			}
+
+			var revealedMap = new RevealedMap(mapId, visible, percentage);
 			conn.Account.AddRevealedMap(revealedMap);
 		}
 
 		/// <summary>
 		/// Reports to the server a percentage of the map that has been explored.
+		/// This packet is seemingly no longer used, it's been combined with the above
```

Comment:
This should've probably rather been a remark than a part of the summary, as it's unrelated to the function of the method. The summary of `CZ_MAP_REVEAL_INFO` meanwhile could've used an update, since it's not correct anymore. Both of these summaries and any potential remarks should ideally also document this transition to a degree and reference each other, in support of future efforts to add backwards compatibility.